bin/faucet/src/server/pow.rs

```diff
@@ -1,16 +1,101 @@
-use std::time::{SystemTime, UNIX_EPOCH};
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+    time::{SystemTime, UNIX_EPOCH},
+};
 
 use axum::{Json, extract::State, response::IntoResponse};
 use miden_tx::utils::ToHex;
 use rand::{Rng, rng};
 use serde::Serialize;
 use sha3::{Digest, Sha3_256};
+use tokio::time::{Duration, interval};
 
 use super::{Server, get_tokens::InvalidRequest};
 
 const DIFFICULTY: u64 = 5;
 const SERVER_TIMESTAMP_TOLERANCE_SECONDS: u64 = 30;
 
+#[derive(Clone)]
+pub struct Challenge {
+    timestamp: u64,
+    server_signature: String,
+}
+
+#[derive(Clone)]
+pub struct ChallengeState {
+    challenges: Arc<Mutex<HashMap<String, Challenge>>>,
+}
+
+impl ChallengeState {
+    pub fn new() -> Self {
```

Comment:
This could just be provided by `#[derive(Default)]` instead. Unless you plan to modify it.